torch/csrc/jit/operator.cpp

```diff
@@ -62,8 +62,14 @@ struct SchemaParser {
     auto tok = L.expect(TK_IDENT);
     auto text = tok.text();
     auto it = type_map.find(text);
-    if(it == type_map.end())
+    if(it == type_map.end()) {
+      if(text.size() > 0 && islower(text[0])) {
+        // lower case identifiers that are not otherwise valid types
+        // are treated as type variables
```

Comment:
If our type vocabulary is fixed, then we really can start treating anything that doesn't match them as type variables. I'm slightly worried about types like `flost` becoming type variables (which trivially match anything).

Reply:
Yeah, the reason I went with lower-case only is to try to avoid typo situation for as much of the identifiers as possible. It is not really possible to avoid for lower-case primitives without even stranger restrictions, or Java-style syntax where Type variables are declared up front. I think the most common failure mode is to forget the name of a non-obvious type (e.g. `RandomSeed` rather than `Generator`).